partcad/src/partcad/shape.py

```
@@ -506,20 +535,43 @@ async def render_3mf_async(
+                    os.path.abspath(filepath),
+                ],
+                request_serialized,
+            )
+            sys.stderr.write(errors)
+
+            response = base64.b64decode(response_serialized)
+            result = pickle.loads(response)
 
-            await pc_thread.run(do_render_3mf)
+            if not result["success"]:
+                pc_logging.error("Render3MF failed: %s: %s" % (self.name, result["exception"]))
```

Ruff (0.8.2)

Use format specifiers instead of percent format

Replace with format specifiers

(UP031)